builders/toolkit/ethereum-api/precompiles/cross-chain-transfers.md

```diff
@@ -0,0 +1,229 @@
+
+The communication protocol enabling token transfers is built on [Substrate](/learn/framework/overview/#substrate-framework){target=\_blank} and runs on a lower level than the EVM, making it harder for EVM developers to access.
+
+This precompile fills the gap between execution layers, exposing a smart contract that abstracts away the underlying complexities, making the execution of cross-chain token transfers as easy as any other smart contract call.
+
+This guide will show you how to interact with the [XCM interface](https://github.com/moondance-labs/tanssi/blob/master/test/contracts/solidity/XcmInterface.sol){target=\_blank} precompile to execute cross-chain token transfers.
+
+The precompile is located at the following address:
+
+```text
+{{networks.dancebox.precompiles.xcmInterface }}
+```
+
+--8<-- 'text/builders/toolkit/ethereum-api/precompiles/security-note.md'
+
+## The XCM Solidity Interface {: #the-xcm-solidity-interface }
+
+The [`XCMInterface.sol`](https://github.com/moondance-labs/tanssi/blob/master/test/contracts/solidity/XcmInterface.sol){target=\_blank} interface on Tanssi EVM appchains is a Solidity interface that allows developers to interact with the precompile's functions.
+
+??? code "XcmInterface.sol"
+
+    ```solidity
+    --8<-- 'code/builders/toolkit/ethereum-api/precompiles/xcm-interface/XcmInterface.sol'
+    ```
+
+The interface includes the necessary data structures along with the following functions:
+
+???+ function "**transferAssetsToPara20**(*uint32* paraId, *address* beneficiary, *AssetAddressInfo[] memory* assets, *uint32* feeAssetItem, *Weight memory* weight) — sends assets to another EVM-compatible appchain using the underlying transfer_assets() transaction included in the module called pallet-xcm"
+
+    === "Parameters"
+
+        - `paraId` - the destination's appchain ID
+        - `beneficiary` - the ECDSA-type account in the destination chain that will receive the tokens
+        - `assets` - an array of assets to send
+        - `feeAssetItem` - the index of the asset that will be used to pay fees
+        - `weight` - the maximum gas to use in the whole operation. Setting uint64::MAX to `refTime` acts in practice as *unlimited weight*
+
+    === "Example"
+
+        - `paraId` - 888
+        - `beneficiary` - 0x3f0Aef9Bd799F1291b80376aD57530D353ab0217
+        - `assets` - [["0x0000000000000000000000000000000000000800", 1000000000000000000]]
+        - `feeAssetItem` - 0
+        - `weight` - [9223372036854775807, 9223372036854775807]
+
+??? function "**transferAssetsToPara32**(*uint32* paraId, *bytes32* beneficiary, *AssetAddressInfo[] memory* assets, *uint32* feeAssetItem, *Weight memory* weight) — sends assets to a Substrate appchain using the underlying transfer_assets() transaction included in the module called pallet-xcm"
```

Comment:
Same comment as above, unsure if we do something with `transfer_assets()` check other functions